apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V57_preadj_add_rbac_groups_and_roles.sql

```diff
@@ -0,0 +1,128 @@
+/*
+    This migration is the first step in a larger initiative that aims to harden our database permissions and to implement
+    least-priviledged access across all schemas.
+
+    This migration:
+        1. Adds a new set of explicitly defined reader, writer, and migrator roles for the pre_adj schema.
+        2. Adds new paca groups to help us manage individual user access:
+            - paca_analyst_group (read only)
+            - paca_data_admins_group (read+write but no alter/drop)
+            - paca_db_admins_group (read+write+alter/drop)
+        3. Sets default permissions so the above gets applied to newly created objects.
+*/
+
+${logic.perms} DO $$
+${logic.perms}  -- new paca roles on pre_adj schema: paca_reader_role, paca_writer_role, paca_migrator_role (read+write+alter)
+${logic.perms}  
+${logic.perms}  -- reader role
+${logic.perms}  IF NOT EXISTS (SELECT FROM pg_catalog.pg_roles WHERE rolname = 'paca_reader_role') THEN
```

Comment:
I don't expect any of these to exist unless we had created them before out of band. But if they did already exist somehow with the wrong privileges, this would not correct that situation. So I'm thinking that it might be more appropriate to 'drop role paca_reader_role if exists' (assuming that is a thing) and then create the role with the privileges that we want.

Reply:
They do not exist, but that's a good call. I will add that for all the others as well.

Reply:
I'm recreating the roles now, but not the groups as the group membership would be dropped in the process.

I can look into a future migration to enforce group options (NOLOGIN, NOINHERIT, etc), but currently a simple `ALTER GROUP WITH` statement does not work because we are not true superusers in our hosted postgres. E.g., we cannot alter `SUPERUSER` or `REPLICATION` properties.